temporalio/worker.py

```diff
@@ -409,22 +410,27 @@ def _heartbeat_activity(self, task_token: bytes, *details: Any) -> None:
         logger = temporalio.activity.logger
         activity = self._running_activities.get(task_token)
         if activity and not activity.done:
-            # Just set as next pending if one is already running
-            coro = self._heartbeat_activity_async(
-                logger, activity, task_token, *details
+            # Put on queue and schedule a task. We will let the queue-full error
+            # be thrown here
+            activity.pending_heartbeats.put_nowait(details)
```

Comment:
We should probably wrap this error with a meaningful error so users can understand this happens because they heartbeat too fast.